src/ckb-daemon/usb_mac.c

```diff
@@ -271,6 +271,67 @@ static void pipecomplete(void* refcon, IOReturn result, void* arg0){
     (*handle)->ReadPipeAsync(handle, ctx->pipe, buffer, ctx->maxsize, pipecomplete, ctx);
 }
 
+CFMachPortRef event_tap;
+
+CGEventRef cg_callback(CGEventTapProxy proxy, CGEventType type, CGEventRef event, void* refcon) {
+    if (type == kCGEventTapDisabledByTimeout) {
+        CGEventTapEnable(event_tap, true);
+        return event;
+    }
+
+    if (event) {
+        for(int i = 0; i < DEV_MAX; i++){
+            if(IS_CONNECTED(keyboard + i)){
+                usbdevice *kb = keyboard + i;
+                if (!IS_MOUSE_DEV(kb)) {
+                    pthread_mutex_lock(imutex(kb));
+                    CGEventSetFlags(event, (kCGEventFlagMaskNonCoalesced | kb->modifiers));
+                    pthread_mutex_unlock(imutex(kb));
+                    break;
+                }
+            }
+        }
+    }
+    return event;
+}
+
+void registerCGEventTap(CFRunLoopRef run_loop) {
+
+    CGEventMask mask = CGEventMaskBit(kCGEventLeftMouseDown) |
+    CGEventMaskBit(kCGEventLeftMouseUp) |
+    CGEventMaskBit(kCGEventRightMouseDown) |
+    CGEventMaskBit(kCGEventRightMouseUp) |
+    CGEventMaskBit(kCGEventMouseMoved) |
+    CGEventMaskBit(kCGEventLeftMouseDragged) |
+    CGEventMaskBit(kCGEventRightMouseDragged) |
+    CGEventMaskBit(kCGEventScrollWheel) |
+    CGEventMaskBit(kCGEventTabletPointer) |
+    CGEventMaskBit(kCGEventTabletProximity) |
+    CGEventMaskBit(kCGEventOtherMouseDown) |
+    CGEventMaskBit(kCGEventOtherMouseUp) |
+    CGEventMaskBit(kCGEventOtherMouseDragged);
```

Comment:
I would prefer the mask-bits to be indented (either under the top mask-bit or all of them by two additional indents). This is just a suggestion, as it is a bit hard to make out the intention of this code block at a brief look.